driver:
  ros__parameters:
    lidar_ip: 169.254.87.155
    computer_ip: 169.254.3.69
    lidar_mode: "512x10"
    imu_port: 7503
    lidar_port: 7502
    sensor_frame: laser_sensor_frame
    laser_frame: laser_data_frame
    imu_frame: imu_data_frame

    # if False, data are published with sensor data QoS. This is preferrable
    # for production but default QoS is needed for rosbag.
    # See: https://github.com/ros2/rosbag2/issues/125
    use_system_default_qos: False

    # Set the method used to timestamp measurements.
    # Valid modes are:
    #
    #   TIME_FROM_INTERNAL_OSC
    #   TIME_FROM_SYNC_PULSE_IN
    #   TIME_FROM_PTP_1588
    #   TIME_FROM_ROS_RECEPTION
    #
    # (See this project's README and/or the Ouster Software Guide for more
    # information).
    #
    timestamp_mode: TIME_FROM_ROS_TIME

    # Mask-like-string used to define the data processors that should be
    # activated upon startup of the driver. This will determine the topics
    # that are available for client applications to consume. The defacto
    # reference for these values are defined in:
    # `include/ros2_ouster/processors/processor_factories.hpp`
    #
    # For convenience, the available data processors are:
    #
    # IMG   - Provides 8-bit image topics suitable for ML applications encoding
    #         the range, ambient and intensity data from a scan
    # PCL   - Provides a point cloud encoding of a LiDAR scan
    # IMU   - Provides a data stream from the LiDARs integral IMU
    # SCAN  - Provides a synthesized 2D LaserScan from the 3D LiDAR data
    #
    # To construct a valid string for this parameter join the tokens from above
    # (in any combination) with the pipe character. For example, valid strings
    # include (but are not limited to):
    #
    # IMG|PCL
    # IMG|PCL|IMU|SCAN
    # PCL
    #
    proc_mask: PCL|IMU|RAW

MCAPRecorderComponent:
  ros__parameters:
    topic: /points
    output_base: /home/ben/lidar_bike_ros_components/test


v4l2_camera:
  ros__parameters:
    image_size: [1920, 1080]

ublox_gps:
  ros__parameters:
    debug: 0                    # Range 0-4 (0 means no debug statements will print)
    device: /dev/ttyACM0
    frame_id: gps
    rate: 10.0
    nav_rate: 10
    uart1:
      baudrate: 115200
    # TMODE3 Config
    TMODE3: 1                   # Survey-In Mode
    sv_in:
      reset: True               # True: disables and re-enables survey-in (resets)
                                # False: Disables survey-in only if TMODE3 is
                                # disabled
      min_dur: 300              # Survey-In Minimum Duration [s]
      acc_lim: 3.0              # Survey-In Accuracy Limit [m]

    inf:
      all: true                   # Whether to display all INF messages in console
    
    publish:
      all: true
      aid:
        hui: false
      nav:
        posecef: false


ublox_node:
  ros__parameters:
    debug: 0                    # Range 0-4 (0 means no debug statements will print)
    device: /dev/ttyACM0
    frame_id: gps
    rate: 10.0
    nav_rate: 1
    uart1:
      baudrate: 115200
    # TMODE3 Config
    tmode3: 0                   # Survey-In Mode
    sv_in:
      reset: True               # True: disables and re-enables survey-in (resets)
                                # False: Disables survey-in only if TMODE3 is
                                # disabled
      min_dur: 300              # Survey-In Minimum Duration [s]
      acc_lim: 3.0              # Survey-In Accuracy Limit [m]

    inf:
      all: true                   # Whether to display all INF messages in console
    
    publish:
      all: true
      aid:
        hui: false
      nav:
        posecef: false

player:
  ros__parameters:
    play:
      # read_ahead_queue_size: 1000
      node_prefix: ""
      rate: 1.0
      loop: false
      # # Negative timestamps will make the playback to not stop.
      # playback_duration:
      #   sec: -1
      #   nsec: 00000000
      start_paused: false
    storage:
      uri: "/home/ben/lb_ros_components/lb_rosbag_1738674363/lb_rosbag_1738674363_0.mcap"
      storage_id: "mcap"
      storage_config_uri: ""

calib_node:
  ros__parameters:
    camera_matrix_row_major: [1.16315864e+03, 0.00000000e+00, 2.80054226e+02, 0.00000000e+00, 1.13928846e+03, 1.86520649e+02, 0.00000000e+00, 0.00000000e+00, 1.00000000e+00]
    image_frame: "camera"
    lidar_frame: "lidar"
